Gems/Atom/Tools/MaterialCanvas/Code/Source/MaterialCanvasApplication.cpp

```diff
@@ -0,0 +1,105 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+
+#include <AtomToolsFramework/Document/AtomToolsDocumentSystemRequestBus.h>
+#include <AzCore/Settings/SettingsRegistryMergeUtils.h>
+#include <MaterialCanvasApplication.h>
+
+#include <Document/MaterialCanvasDocument.h>
+#include <Window/MaterialCanvasGraphView.h>
+#include <Window/MaterialCanvasMainWindow.h>
+
+void InitMaterialCanvasResources()
+{
+    // Must register qt resources from other modules
+    Q_INIT_RESOURCE(MaterialCanvas);
+    Q_INIT_RESOURCE(InspectorWidget);
+    Q_INIT_RESOURCE(AtomToolsAssetBrowser);
+}
+
+namespace MaterialCanvas
+{
+    static const char* GetBuildTargetName()
+    {
+#if !defined(LY_CMAKE_TARGET)
+#error "LY_CMAKE_TARGET must be defined in order to add this source file to a CMake executable target"
+#endif
+        return LY_CMAKE_TARGET;
+    }
+
+    MaterialCanvasApplication::MaterialCanvasApplication(int* argc, char*** argv)
+        : Base(GetBuildTargetName(), argc, argv)
+    {
+        InitMaterialCanvasResources();
+
+        QApplication::setOrganizationName("O3DE");
+        QApplication::setApplicationName("O3DE Material Canvas");
```

Comment:
I suggest we label this "O3DE Material Canvas PROTOTYPE WIP" or something like that until it's in a usable state.